src/libMessage/Messenger.cpp

```diff
@@ -1240,24 +1240,19 @@ bool ProtobufToTransaction(const ProtoTransaction& protoTransaction,
     return false;
   }
 
-  SHA2<HashType::HASH_VARIANT_256> sha2;
```

Comment:
The code for checking TX hash was outside of TX construction here - that seemed wrong/unclean. Let the TX create its own hash then compare that to what we think it should be.